openlibrary/plugins/worksearch/code.py

```diff
@@ -636,16 +636,41 @@ class list_search(delegate.page):
     path = '/search/lists'
 
     def GET(self):
+        i = web.input(q='', offset='0', limit='10')
+
+        lists = self.get_results(i.q, i.offset, i.limit)
+
+        return render_template('search/lists.tmpl', q=i.q, lists=lists)
+
+    def get_results(self, q, offset=0, limit=100):
         if 'env' not in web.ctx:
             delegate.fakeload()
 
-        i = web.input(q='', offset='0', limit='10')
         keys = web.ctx.site.things({
-            "type": "/type/list", "name~": i.q,
-            "limit": int(i.limit), "offset": int(i.offset)
+            "type": "/type/list", "name~": q,
+            "limit": int(limit), "offset": int(offset)
         })
-        lists = web.ctx.site.get_many(keys)
-        return render_template('search/lists.tmpl', q=i.q, lists=lists)
+
+        return web.ctx.site.get_many(keys)
+
+class list_search_json(list_search):
+    path = '/search/lists'
+    encoding = 'json'
+
+    def GET(self):
+        i = web.input(q='', offset=0, limit=10)
+        offset = safeint(i.offset, 0)
+        limit = safeint(i.limit, 10)
+        limit = min(100, limit)  # limit limit to 1000.
+
+        docs = self.get_results(i.q, offset=offset, limit=limit)
+
+        response = {
+            'docs': [doc.preview() for doc in docs]
```

Comment:
The subjects search endpoint includes a `start` field which displays the offset; let's do that here as well.

Reply:
Ah, so it does.  It also includes a `numFound` - for SOLR that's not too bad to query, but in this case it may be, given that it's a SQL query.

What do you think - should we do that `COUNT` operation anyway?

Reply:
Scratch that - I don't think a `COUNT` is possible because we aren't directly hitting any database - we're calling a `things` endpoint, it looks like?

Reply:
Yeah, that's what I was thinking. We could try to do a count query, but I think we'd run into performance issues. So leaving it out seems reasonable.